tests/toranj/test-023-multicast-traffic.py

```diff
@@ -0,0 +1,203 @@
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+
+import time
+import wpan
+from wpan import verify
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Test description: Multicast traffic
+#
+# Network topology
+#
+#     r1 ---- r2 ---- r3 ---- r4
+#             |               |
+#             |               |
+#            fed             sed
+#
+# Test covers the following multicast traffic:
+#
+# - r2  =>> link-local all-nodes.   Expected to receive on [r1, r2, r3, fed].
+# - r3  =>> mesh-local all-nodes.   Expected to receive on [r1, r2, r3, r4, fed].
+# - r3  =>> link-local all-routers. Expected to receive on [r2, r3, r4].
+# - r3  =>> mesh-local all-routers. Expected to receive on all routers.
+# - r1  =>> link-local all-thread.  Expected to receive on [r1, r2].
+# - fed =>> mesh-local all-thread.  Expected to receive on all nodes.
+# - r1  =>> specific address (on r2 and sed). Expected to receive on [r2, sed].
+#
+
+test_name = __file__[:-3] if __file__.endswith('.py') else __file__
+print '-' * 120
+print 'Starting \'{}\''.format(test_name)
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Utility functions
+
+def send_mcast(src_node, src_addr, mcast_addr, recving_node, msg_len=30):
+    """
+    Send a multicast message with given `len` from `src_node` using `src_addr` to the multicast address `mcast_addr`.
+    Verify that the message is received on all nodes in `recving_node` list.
+    """
+    sender = src_node.prepare_tx(src_addr, mcast_addr, msg_len, mcast_hops=5)
```

Comment:
Should we also validate the hop limit?

Reply:
Same situation as above.

I know it works, since default hop limit seems to be 1 and without setting it, I noticed that traffic going only one hop...